api/src/main/java/org/openmrs/module/kenyaemr/advice/SyncHFEAppointmentsWithBahmniModule.java

```diff
@@ -95,56 +95,60 @@ public class SyncHFEAppointmentsWithBahmniModule implements AfterReturningAdvice
-                    hivFollowUpAppointment.setDateVoided(new Date());
-                    hivFollowUpAppointment.setVoidedBy(Context.getAuthenticatedUser());
-                }
+                    if(nxtAppointment && followUpAppointment) {
+                        Appointment app = appointmentsService.validateAndSave(hivFollowUpAppointment);
+                    }
 
-                if(nxtAppointment && followUpAppointment) {
-                    Appointment app = appointmentsService.validateAndSave(hivFollowUpAppointment);
-                }
+                    // delete existing appointment for drug refill
+                    if(drugRefillAppointment != null) {
+                        drugRefillAppointment.setVoided(true);
+                        drugRefillAppointment.setDateVoided(new Date());
+                        drugRefillAppointment.setVoidedBy(Context.getAuthenticatedUser());
+                        Appointment app = appointmentsService.validateAndSave(drugRefillAppointment);
+                    }
 
-                // delete existing appointment for drug refill
-                if(drugRefillAppointment != null) {
-                    drugRefillAppointment.setVoided(true);
-                    drugRefillAppointment.setDateVoided(new Date());
-                    drugRefillAppointment.setVoidedBy(Context.getAuthenticatedUser());
-                    Appointment app = appointmentsService.validateAndSave(drugRefillAppointment);
+                } else if (editAppointment != null && enc.getForm() != null &&
+                        (enc.getForm().getUuid().equals(HivMetadata._Form.HIV_GREEN_CARD) || enc.getForm().getUuid().equals(HivMetadata._Form.MOH_257_VISIT_SUMMARY))) {
+                    // pick HIV followup forms
+                    processEditHivFollowupEncounter(enc);
+
+                } else if (enc != null && enc.getForm() != null &&
+                        (enc.getForm().getUuid().equals(HivMetadata._Form.HIV_GREEN_CARD) || enc.getForm().getUuid().equals(HivMetadata._Form.MOH_257_VISIT_SUMMARY))) {
+                    // pick HIV followup forms
+                    processCreateHivFollowupEncounter(enc);
+
+                } else if(enc != null && enc.getForm() != null &&
+                        (enc.getForm().getUuid().equals(MchMetadata._Form.MCHMS_ANTENATAL_VISIT) ||
+                                enc.getForm().getUuid().equals(MchMetadata._Form.MCHCS_FOLLOW_UP) ||
+                                enc.getForm().getUuid().equals(PREP_FOLLOWUP_FORM) ||
+                                enc.getForm().getUuid().equals(PREP_INITIAL_FORM) ||
+                                enc.getForm().getUuid().equals(PREP_MONTHLY_REFILL_FORM) ||
+                                enc.getForm().getUuid().equals(KP_CLINICAL_VISIT_FORM) ||
+                                enc.getForm().getUuid().equals(TbMetadata._Form.TB_FOLLOW_UP) )) {
+
+                    processProgramEncounter(enc);
```

Comment:
can we edit this method to reflect its role, or what is program encounter? We can be explicit